docs/designs/ps_design.md

```diff
@@ -1,53 +1,62 @@
 
-The master will create a PS with *N* PS nodes, where *N* is specified by the user. Each model variable and embedding vector has a corresponding PS node. Thus, every PS node has a subset of model variables and embedding tables.
+A typical parameter server architecture of data parallelism distributed training system contains two roles:
 
-The master will monitor PS nodes status similar to what it does for workers. In case a PS node fails, the master will try to relaunch it. Since PS nodes have higher priority than workers, if there are still some running workers, the relaunch will succeed by using either idle or preempted Kubernetes resources. If the relaunch fails, there are no workers left. The Elastic job has to wait for resources for the PS node and workers.
+- parameter server(or PS for short), maintaining the global model, and providing parameter pull/push/optimize/checkpoint service
+- worker, computing gradients of model parameters using local data and pushing gradients to parameter servers.
 
-Each worker has a local copy of the model variables. After the master relaunches a PS node, the PS node can recover model variables from workers. For embedding vectors, PS must create replicas to support fault tolerance. For each PS node *PS(i)*, it will store *M* replicas in the following *M* PS nodes from *PS(i+1 % N)* to *PS(i+M % N)*. The relaunched PS node can recover embedding vectors from one of its replicas. If there are more than *M* continuously-indexed PS nodes failing, at least one PS node fails with all of its replicas. The ElasticDL job has to recover from a recent checkpoint.
+Sharding the model to multiple parameter servers is a common choice. The reasons are as following:
 
-## PS
+- Sometimes, the model size if out of the memory of a process. For example, many recommending and ranking models take very high-dimensional (and sparse) inputs, thus require large embedding tables.
+- Communication load between workers and parameter servers will be balanced. More parameter servers usually means more network bandwidth.
+- Model optimization could be parallelized on multiple parameter servers.
 
-### Compoments
+In addition, we want to launch machine learning jobs in a Kubernetes supported cluster. The pods in Kubernetes are scheduled with priority, and could be preempted at any time. At the same time, hardware failure problem is also nonnegligible in a large scale distributed system. Here, we introduce another role called master to watch the status of worker pods and parameter server pods. It calls Kubernetes API to start or relaunch worker pods and optionally parameter server pods, and scheduling tasks to worker pods.
```

Comment:
I understand this paragraph has an intention to explain the existence of the master process in a job. To clearly explain this topic, the author needs to know more about Kubernetes-native architecture and its successor, the Kubernetes Operator-based design strategy. Otherwise, it is difficult to explain why we need a master process, other than a Kubernetes Operator.

Again, do we need to explain why we need a master here?